packages/config/src/android/Updates.ts

```diff
@@ -0,0 +1,103 @@
+import { ExpoConfig } from '../Config.types';
+import { Document } from './Manifest';
+
+export function getUpdateUrl(config: ExpoConfig, username: string | null) {
+  const user = typeof config.owner === 'string' ? config.owner : username;
+  if (!user) {
+    return null;
+  }
+  return `https://exp.host/@${user}/${config.slug}`;
+}
+
+export function getSDKVersion(config: ExpoConfig) {
+  return typeof config.sdkVersion === 'string' ? config.sdkVersion : null;
+}
+
+export function getUpdatesEnabled(config: ExpoConfig) {
+  return config.updates?.enabled !== false;
+}
+
+export function getUpdatesTimeout(config: ExpoConfig) {
+  return config.updates?.fallbackToCacheTimeout ?? 0;
+}
+
+export function getUpdatesCheckOnLaunch(config: ExpoConfig) {
+  if (config.updates?.checkAutomatically === 'ON_ERROR_RECOVERY') {
+    return 'NEVER';
+  } else if (config.updates?.checkAutomatically === 'ON_LOAD') {
+    return 'ALWAYS';
+  }
+  return 'ALWAYS';
+}
+
+export async function setUpdatesConfig(
+  config: ExpoConfig,
+  manifestDocument: Document,
+  username: string | null
+) {
+  let mainApplication = manifestDocument.manifest.application.filter(
+    (e: any) => e['$']['android:name'] === '.MainApplication'
+  )[0];
+
+  addMetaDataItemToMainApplication(
+    mainApplication,
+    'expo.modules.updates.ENABLED',
+    `${getUpdatesEnabled(config)}`
+  );
+  addMetaDataItemToMainApplication(
+    mainApplication,
+    'expo.modules.updates.EXPO_UPDATES_CHECK_ON_LAUNCH',
+    getUpdatesCheckOnLaunch(config)
+  );
+  addMetaDataItemToMainApplication(
+    mainApplication,
+    'expo.modules.updates.EXPO_UPDATES_LAUNCH_WAIT_MS',
+    `${getUpdatesTimeout(config)}`
+  );
+
+  const updateUrl = getUpdateUrl(config, username);
+  if (updateUrl) {
+    addMetaDataItemToMainApplication(
+      mainApplication,
+      'expo.modules.updates.EXPO_UPDATE_URL',
+      updateUrl
+    );
+  }
+  const sdkVersion = getSDKVersion(config);
+  if (sdkVersion) {
+    addMetaDataItemToMainApplication(
+      mainApplication,
+      'expo.modules.updates.EXPO_SDK_VERSION',
+      sdkVersion
+    );
+  }
+
+  return manifestDocument;
+}
+
+function addMetaDataItemToMainApplication(
```

Comment:
nice idea to extract this out, think there are a couple other places in the `config/android/` source that could make use of this ([like here](https://github.com/expo/expo-cli/blob/master/packages/config/src/android/Branch.ts#L19-L37)) (not that you need to go through and make those changes now)

Reply:
think it's worth moving this util to `Manifest.ts`?

Reply:
(also, full disclosure, I stole the idea from `Facebook.ts` :) )

Reply:
haha 😄 
Yeah I think it's worth it, but no need to replace the duplicate code in the other files' `set` methods (unless you feel very inclined to do so; I think the relevant unit tests would catch any problems)